src/app/armory/weapon-roll-export.ts

```diff
@@ -0,0 +1,101 @@
+import { DimItem } from 'app/inventory/item-types';
+import {
+  getSocketsWithStyle,
+  getWeaponArchetypeSocket,
+  isWeaponMasterworkSocket,
+} from 'app/utils/socket-utils';
+import { DestinySocketCategoryStyle } from 'bungie-api-ts/destiny2';
+import { ItemCategoryHashes } from 'data/d2/generated-enums';
+
+/**
+ * Exports a weapon roll as human-readable text format suitable for sharing
+ * on platforms like /r/sharditkeepit
+ */
+export function formatWeaponRollForExport(item: DimItem): string {
+  if (!item.bucket.inWeapons || !item.sockets) {
+    return '';
+  }
+
+  const lines: string[] = [];
+
+  // Weapon name
+  lines.push(`[${item.name}]`);
+  lines.push('');
+
+  // Intrinsic Perk / Frame (Archetype)
+  const archetypeSocket = getWeaponArchetypeSocket(item);
+  if (archetypeSocket?.plugged?.plugDef.displayProperties.name) {
+    lines.push(`* ${archetypeSocket.plugged.plugDef.displayProperties.name}`);
+  }
+
+  // Sights/Barrels - these are typically large perks
+  const largePerks = getSocketsWithStyle(item.sockets, DestinySocketCategoryStyle.LargePerk).filter(
+    (socket) => socket.plugged?.plugDef.displayProperties.name,
+  );
+
+  if (largePerks.length > 0) {
+    for (const socket of largePerks) {
+      lines.push(`* ${socket.plugged!.plugDef.displayProperties.name}`);
+    }
+  }
+
+  // Trait columns (reusable perks) - excluding weapon mods and masterwork
+  const reusablePerks = getSocketsWithStyle(
+    item.sockets,
+    DestinySocketCategoryStyle.Reusable,
+  ).filter((socket) => {
+    if (!socket.plugged?.plugDef.displayProperties.name) {
+      return false;
+    }
+
+    // Skip weapon mods
+    if (socket.plugged.plugDef.itemCategoryHashes?.includes(ItemCategoryHashes.WeaponModsDamage)) {
+      return false;
+    }
+
+    // Skip masterwork sockets
+    return !isWeaponMasterworkSocket(socket);
+  });
+
+  for (const socket of reusablePerks) {
+    lines.push(`* ${socket.plugged!.plugDef.displayProperties.name}`);
+  }
```

Comment:
We're printing the whole roll here, not just the plugged plug. Print out all the plugOptions separated by `/`